src_c/math.c

```diff
@@ -4088,12 +4088,17 @@ static PyObject *
 math_clamp(PyObject *self, PyObject *const *args, Py_ssize_t nargs)
 {
     if (nargs != 3)
-        return RAISE(PyExc_ValueError, "clamp requires 3 arguments");
+        return RAISE(PyExc_TypeError, "clamp requires 3 arguments");
 
     PyObject *value = args[0];
     PyObject *min = args[1];
     PyObject *max = args[2];
 
+    if (PyNumber_Check(args[0]) != 1 || PyNumber_Check(args[1]) != 1 ||
+        PyNumber_Check(args[2]) != 1) {
+        return RAISE(PyExc_TypeError, "clamp requires 3 numeric arguments");
+    }
+
     // if value < min: return min
     int result = PyObject_RichCompareBool(value, min, Py_LT);
```

Comment:
I know this isn't the focus of this PR, but it would be cool to put in a comment that the reason the function is implemented without full conversion into C types is for performance reasons, and this was tested to be faster.